### DISCUSSION TITLE
Separate DB for webapp tables and user tables

### DISCUSSION THREAD

**Discussion by mathemancer** [mathesar-foundation/mathesar#23] [Maintainer discussions]
Brought over from #mathesar:matrix.org.  

After getting the email type set up and working, I think we should consider using a separate database for the user tables than the Django tables, even in the single-user setup. This would make a few things a bit clearer, and also avoid any problems when the user (inevitably) tries to import a bunch of tables from the public schema into the app. I think we should always create a separate database for Mathesar Django tables that's completely owned by the web app.
Pros:
- clearer distinction between user and web app tables (obviously)
- Easier testing, we can avoid Django getting in the way for testing SQLAlchemy functions
- Easier extension to multi-user installations
- More "proper"  (the web app really should own its own DB that shouldn't be modified by the user under most circumstances).
Cons:
- Slightly more complex.
- need to figure out user management sooner rather than later

The second con above could be mitigated for now by just setting up two DBs in the Django settings, but I'm not sure it's a good idea to have Django managing the connection string info for databases the web app doesn't own.

**Comment by kgodey** [comment]
I agree that it makes sense to have a separate database for the web app, for all the reasons you listed.

We're currently using Django settings for configuring everything related to Mathesar, I think it makes sense to set up the second DB in the Django settings for now to avoid having multiple places to set up configuration. We'll need to make some changes to that set up pretty soon, since we'll need the ability for the user to log in with a postgres user and view all their databases.

**Comment by mathemancer** [comment]
This is solved in #49
